ge/org.osate.ge.errormodel/src/org/osate/ge/errormodel/ErrorBehaviorStateHandler.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Created, in part, with funding and support from the United States Government. (see Acknowledgments file).
+ *
+ * This program includes and/or can make use of certain third party source code, object code, documentation and other
+ * files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+ * configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+ * conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.errormodel;
+
+import java.util.Optional;
+
+import org.osate.aadl2.AadlPackage;
+import org.osate.ge.CanonicalBusinessObjectReference;
+import org.osate.ge.GraphicalConfiguration;
+import org.osate.ge.GraphicalConfigurationBuilder;
+import org.osate.ge.RelativeBusinessObjectReference;
+import org.osate.ge.businessobjecthandling.BusinessObjectHandler;
+import org.osate.ge.businessobjecthandling.CanDeleteContext;
+import org.osate.ge.businessobjecthandling.CanRenameContext;
+import org.osate.ge.businessobjecthandling.GetGraphicalConfigurationContext;
+import org.osate.ge.businessobjecthandling.GetNameContext;
+import org.osate.ge.businessobjecthandling.IsApplicableContext;
+import org.osate.ge.businessobjecthandling.ReferenceContext;
+import org.osate.ge.businessobjecthandling.RenameContext;
+import org.osate.ge.errormodel.util.ErrorModelGeUtil;
+import org.osate.ge.errormodel.util.ErrorModelNamingUtil;
+import org.osate.ge.graphics.EllipseBuilder;
+import org.osate.ge.graphics.Graphic;
+import org.osate.xtext.aadl2.errormodel.errorModel.ErrorBehaviorState;
+import org.osate.xtext.aadl2.errormodel.errorModel.ErrorBehaviorStateMachine;
+
+public class ErrorBehaviorStateHandler implements BusinessObjectHandler {
+	private static final Graphic graphic = EllipseBuilder.create().build();
+
+	@Override
+	public boolean isApplicable(final IsApplicableContext ctx) {
+		return ctx.getBusinessObject(ErrorBehaviorState.class).map(bo -> bo.getElementRoot() instanceof AadlPackage)
+				.isPresent();
+	}
+
+	@Override
+	public CanonicalBusinessObjectReference getCanonicalReference(final ReferenceContext ctx) {
+		final ErrorBehaviorState typedBo = ctx.getBusinessObject(ErrorBehaviorState.class).get();
```

Comment:
Possible handling of exception better, instead of calling get(). 
Could be something like:
````
ctx.getBusinessObject(ErrorBehaviorStateMachine.class).map(sm -> new CanonicalBusinessObjectReference(ErrorModelReferenceUtil.TYPE_BEHAVIOR_STATE_MACHINE,
ctx.getReferenceBuilder().getCanonicalReference(sm.getElementRoot()).encode(), sm.getName()))
.orElseThrow(() -> new RuntimeException("Cannot find canonical reference"));
````

Note: This is being done in quite a few places.  Maybe throw an exception in the getBusinessObject method instead of changing code everywhere.  I would like your input on this as well.

Reply:
An exception needs to thrown if a canonical reference cannot be retrieved. We are assuming that the business object being provided is of the desired type so we are using `get()`. This is a safe assumption because `isApplicable` checks the type and other methods aren't called unless the handler is applicable. Usually it is good idea to avoid `get` but I believe this is one case where it is reasonable to use. If/when OSATE uses Java 11, we'll start using `orElseThrow()` (without arguments) for cases like this. That method is only in Java 11 and was added because they realized that they shouldn't have named it `get`.